apps/managedidentity/managedidentity_test.go

```diff
@@ -465,6 +465,105 @@ func TestIMDSAcquireTokenReturnsTokenSuccess(t *testing.T) {
 	}
 }
 
+func TestCloudShellAcquireTokenReturnsTokenSuccess(t *testing.T) {
+	resource := "https://resource"
+	miType := SystemAssigned()
+
+	setEnvVars(t, CloudShell)
+	endpoint := os.Getenv(msiEndpointEnvVar)
+
+	var localUrl *url.URL
+	var resourceString string
+	mockClient := mock.Client{}
+	responseBody, err := getSuccessfulResponse(resource, false)
+	if err != nil {
+		t.Fatalf(errorFormingJsonResponse, err.Error())
+	}
+
+	mockClient.AppendResponse(mock.WithHTTPStatusCode(http.StatusOK), mock.WithBody(responseBody), mock.WithCallback(func(r *http.Request) {
+		localUrl = r.URL
+		err = r.ParseForm()
+		if err != nil {
+			t.Fatal(err)
+		}
+		resourceString = r.FormValue(resourceQueryParameterName)
+	}))
+
+	// Resetting cache
+	before := cacheManager
+	defer func() { cacheManager = before }()
+	cacheManager = storage.New(nil)
+
+	client, err := New(miType, WithHTTPClient(&mockClient))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	result, err := client.AcquireToken(context.Background(), resource)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if localUrl == nil || !strings.HasPrefix(localUrl.String(), endpoint) {
+		t.Fatalf("url request is not on %s got %s", endpoint, localUrl)
+	}
+
+	if resourceString != strings.TrimSuffix(resource, "/.default") {
+		t.Fatal("suffix /.default was not removed.")
+	}
+
+	if result.Metadata.TokenSource != base.IdentityProvider {
+		t.Fatalf("expected IdentityProvider tokensource, got %d", result.Metadata.TokenSource)
+	}
+
+	if result.AccessToken != token {
+		t.Fatalf("wanted %q, got %q", token, result.AccessToken)
+	}
+
+	result, err = client.AcquireToken(context.Background(), resource)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if result.Metadata.TokenSource != base.Cache {
+		t.Fatalf("wanted cache token source, got %d", result.Metadata.TokenSource)
+	}
+
+	secondFakeClient, err := New(miType, WithHTTPClient(&mockClient))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	result, err = secondFakeClient.AcquireToken(context.Background(), resource)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if result.Metadata.TokenSource != base.Cache {
+		t.Fatalf("cache result wanted cache token source, got %d", result.Metadata.TokenSource)
+	}
+}
+
+func TestCloudShellOnlySystemAssignedSupported(t *testing.T) {
+	setEnvVars(t, CloudShell)
+	mockClient := mock.Client{}
+
+	for _, testCase := range []ID{
+		UserAssignedClientID("client"),
+		UserAssignedObjectID("ObjectId"),
+		UserAssignedResourceID("resourceid")} {
```

Comment:
this is more conventional, though gofmt doesn't require it
```suggestion
		UserAssignedResourceID("resourceid"),
	} {
```